### PR TITLE
check json_encode behavior for vim.empty_dict()

### PR DESCRIPTION
**Pull request by h-michael**

### CODE REVIEW

**Comment by h-michael** [comment]
@justinmk ref: https://github.com/neovim/nvim-lsp/pull/121

I wondered if it was a json_encode problem and tried a test. Is the result of this test a bug, or is the test written badly?

https://travis-ci.org/neovim/neovim/jobs/649375062#L6957-L6965

File: runtime/lua/vim/shared.lua [hunk 1]
```diff
@@ -6,6 +6,10 @@
 
 local vim = vim or {}
 
+function vim.empty_dict()
```
Comment (bfredl) [inline runtime/lua/vim/shared.lua]: This does not make any sense. `vim._empty_dict_mt` exists only in in process lua. You would need to extend libmpack-lua with the corresponding functionality.

File: test/functional/eval/json_functions_spec.lua [hunk 2]
```diff
@@ -569,6 +569,8 @@ describe('json_encode() function', function()
 
   it('dumps dictionaries', function()
     eq('{}', eval('json_encode({})'))
+    eq('{}', funcs.json_encode(helpers.empty_dict()))
+    eq('{"a": {}}', funcs.json_encode({a=helpers.empty_dict()}))
```
Comment (bfredl) [inline test/functional/eval/json_functions_spec.lua]: you probably want `exec_lua([[ return vim.fn.json_encode({a=vim.empty_dict()}) ]])` (and place it in `lua/vim_spec.lua` among the other `empty_dict()` tests)

**Comment by h-michael** [comment]
@bfredl Thanks for correcting my wrong verification.
The behavior of json_encode for empty_dict seems to be correct.

**Comment by bfredl** [comment]
Merged. At some point we should have "native" JSON support in lua, and then these tests would be useful.

**Pull request merged**